### PR TITLE
Site editor: go direct to edit from manage all templates list

### LINKED ISSUE #48702: Manage templates: Try going directly to edit view
Currently, clicking a template in the Manage templates view takes you to the Details panel and updates the preview:

https://user-images.githubusercontent.com/846565/222517331-c54248dd-df89-4512-b848-c897d8697d8e.mp4

This is sub-optimal for a couple of reasons:

1. You can't really do anything in the Details panel yet, aside from editing which makes this step feel superfluous
2. It makes getting back to the Manage Templates view a bit tedious, requiring an extra click

Instead let's try going directly into the edit experience when you click a template title. 

I would then expect clicking the W button to take me back to the template list, solving the second point above. But I appreciate this routing is a bit unorthodox and may need to be investigated separately.

### PR DESCRIPTION
**Pull request by glendaviesnz**
## What?
If clicking on a template in the manage all templates list go direct to the editor rather than  the interim view canvas for the template.

## Why?
The interim view step is currently redundant as offers no additional functionality.
Fixes: #48702

## How?

- Replaces `canvas=view` with `canvas=edit` in the link params
- Adds a `fromTemplateList` param so the back link can be correctly generated in the editor

## Testing Instructions

- Open the site editor and go to the 'Manage all templates' page.
- Click on  a template in the list and check that it goes direct to the editor
- Click on the WordPress icon and check it returns to the 'Manage all templates' page
- Check that the back button works to return from the editor to the 'Manage all templates' page (currently broken)

## Screenshots or screencast 

Before:

https://user-images.githubusercontent.com/3629020/223001747-c99592e7-b9d1-4aa0-8850-f5c0365b9f33.mov

After:

https://user-images.githubusercontent.com/3629020/223001791-05f0d66c-09ec-4722-8c5e-5cd978c352c7.mov

### CODE REVIEW

**Comment by glendaviesnz** [comment]
Just draft at the moment as this breaks the browser back button from the editor back to all templates list. I need to have a look at #48731 and work out if this will fix it, or if I am doing something wrong with the use of `history`

**Comment by glendaviesnz** [comment]
https://github.com/WordPress/gutenberg/pull/48731 does seem to fix the back button issue so we should probably just hold off merging this until that is merged and then I will retest this, but will also have a bit more of a look at it tomorrow to see if there is a smarter way to sort the back link to the manage all templates list without having to pass through the extra param.

**Comment by glendaviesnz** [comment]
It would be nice if there was a way to get the backlink to work without the SiteHub needing to be explicitly aware that the template had been linked to from the all templates list, but I couldn't think of an alternative.

**Comment by youknowriad** [comment]
Personally, I prefer largely the behavior in trunk. I think we're only seeing this as a problem because there's not much you can do yet from the sidebar of the site editor when a template is selected.

**Comment by jameskoster** [comment]
The problem with the behaviour on trunk, in terms of UX:

1. It feels unexpected due to the inconsistency with every other content management list in WordPress, where:
   1. Clicking an item title takes you directly to the full edit experience.
   2. Subsequently, clicking the W button takes you back to list view
2. It makes getting back to the management list rather laborious, particularly when the template list is long and you have to scroll.
3. It creates an awkward Back button experience in the sidebar: after selecting a template from the list view, you might expect the [<] button to return you to the list view, which it does not.

**Comment by youknowriad** [comment]
I understand this but I still believe trunk is better personally.

As seen in this PR, we need to store the "previous" location and change the behavior of buttons depending on that previous location, this is very fragile and make the experience inconsistent. The "W" button does two different things depending on where it's clicked. It's important to note that the "w" button is not a "back" button. Maybe that's how you picture it but I don't think that's what everyone sees.

Again, when we'll have the possibility to change styles .... in the sidebar, I feel this feeling of an unnecessary step is just going to fade away IMO.

**Comment by jameskoster** [comment]
I appreciate the desire to preserve the singular behavior of the W button in terms of the code. I don't actually see it as a back button FWIW, indeed it already has two functions:

1. Zoom out of full screen editing
2. Exit Site Editor / view Dashboard

The tricky part with the first point is what do display when zoomed out. In the Post Editor we 'zoom out' to the list view, which seems right in that context. 

Perhaps we leave the W button until we have a clearer picture of some broader workflows – something we'll be exploring more for Phase 3.

---

To me, it still feels a bit unexpected to see the details interstitial via the Manage list. From the manage list I would expect to go directly to full screen editing... and if I wanted the overview I'd just click Templates > 404 rather than Templates > Manage > 404.

Maybe we _try_:

1. Keep the go-directly-to-edit behaviour
2. Leave the W button behavior as it is on trunk

If that doesn't feel good then we can close :)

**Comment by youknowriad** [comment]
I'm ok with this as a compromise yeah 👍

**Comment by glendaviesnz** [comment]
> * Keep the go-directly-to-edit behaviour
> * Leave the W button behavior as it is on trunk

Done.

File: packages/edit-site/src/components/list/table.js
Comment (youknowriad) [inline packages/edit-site/src/components/list/table.js]: Is this still necessary?

**Comment by mtias** [comment]
Let's just do go-directly-to-edit but don't change behaviour of W.

**Review by jameskoster — APPROVED** [review]
Thanks Glen, this is working well from the design perspective.

File: packages/edit-site/src/components/list/table.js
Comment (glendaviesnz) [reply packages/edit-site/src/components/list/table.js]: 🤦 - gone now.

**Pull request merged**
